lib/sender.js

```diff
@@ -0,0 +1,69 @@
+'use strict';
+
+const net = require('net')
+const tls = require('tls')
+const { Writable } = require('stream')
+
+
+/**
+ * Event sender.
+ * Can be used as a stream, with the usual functions:
+ * sender.write(), sender.end(), etc.
+ * Implements backpressure.
+ */
+class Sender extends Writable {
+
+  /**
+   * Create the client.
+   * @param {Object} options for sender, including:
+   *  - host: the host to send events, default localhost.
+   *  - port: the destination port (mandatory).
+   *  - objectMode: whether to allow sending objects.
+   * @param {function(*)} callback (Optional) Function to invoke when the
+   * connection has been established.
+   */
+  constructor(options, callback) {
+    super(options)
+    this._options = options
+    const libnet = options.cert ? tls : net
+    this._socket = libnet.connect(options, callback)
+    this._socket.on('error', error => this.emit('error', error))
+    this._socket.unref()
+  }
+
+  /**
+   * Send an event to the ingestion endpoint.
+   *
+   * @param {string|Buffer|Object} event message to send.
+   * If an object is passed it will be converted to JSON before sending.
+   * @param {function(*)} callback Optional function to invoke with (error).
+   */
+  send(event, callback) {
+    const converted = this._convert(event)
+    return this._socket.write(converted, null, error => {
```

Comment:
I would omit the encoding or set it as `utf8`

Reply:
I am passing `null` which is not setting it. Would you prefer `undefined`?

Reply:
I think it is better to omit it, like `this._socket.write(converted, error => {...})`